extensions/sql-database-projects/src/dialogs/publishDatabaseQuickpick.ts

```diff
@@ -0,0 +1,152 @@
+		return {
+			label: db,
+			dbName: db
+		} as vscode.QuickPickItem & { dbName: string, isCreateNew?: boolean };
+	});
+	// Ensure the project name is an option, either adding it if it doesn't already exist or moving it to the top if it does
+	const projectNameIndex = dbs.findIndex(db => db === project.projectFileName);
+	if (projectNameIndex === -1) {
+		dbQuickpicks.unshift({ label: constants.newDatabaseTitle(project.projectFileName), dbName: project.projectFileName });
+	} else {
+		dbQuickpicks.splice(projectNameIndex, 1);
+		dbQuickpicks.unshift({ label: project.projectFileName, dbName: project.projectFileName });
+	}
+
+	dbQuickpicks.push({ label: constants.createNew, dbName: '', isCreateNew: true });
+	// 3. Select database
+	// TODO: Hook up to MSSQL
+	let databaseName = '';
+	while (databaseName === '') {
+		const selectedDatabase = await vscode.window.showQuickPick(
+			dbQuickpicks,
+			{ title: constants.selectDatabase, ignoreFocusOut: true });
+		if (!selectedDatabase) {
+			// User cancelled
+			return;
+		}
+		databaseName = selectedDatabase.dbName;
+		if (selectedDatabase.isCreateNew) {
+			databaseName = await vscode.window.showInputBox(
+				{
+					title: constants.enterNewDatabaseName,
+					ignoreFocusOut: true
+				}
+			) ?? '';
+			// If user cancels out of this just return them to the db select quickpick in case they changed their mind
+		}
+	}
+
+
+	// 4. Modify sqlcmd vars
+	// TODO: Concat ones from publish profile
+	let sqlCmdVariables = Object.assign({}, project.sqlCmdVariables);
+
+	if (Object.keys(sqlCmdVariables).length > 0) {
+		// Loop over the vars to modify until the user is done
+		while (true) {
+			const quickPickItems = Object.keys(sqlCmdVariables).map(key => {
+				return {
+					label: key,
+					description: sqlCmdVariables[key],
+					key: key
+				} as vscode.QuickPickItem & { key?: string, isResetAllVars?: boolean, isDone?: boolean };
+			});
+			quickPickItems.push({ label: constants.resetAllVars, isResetAllVars: true });
+			quickPickItems.unshift({ label: constants.done, isDone: true });
+			const sqlCmd = await vscode.window.showQuickPick(
+				quickPickItems,
+				{ title: constants.chooseSqlcmdVarsToModify, ignoreFocusOut: true }
+			);
+			if (!sqlCmd) {
+				// When user hits escape then we continue on here, we don't exit the publish
+				// flow since this is an optional step
+				break;
+			}
+			if (sqlCmd.key) {
+				const newValue = await vscode.window.showInputBox(
+					{
+						title: sqlCmd.key,
+						value: sqlCmdVariables[sqlCmd.key],
+						ignoreFocusOut: true
+					}
+				);
+				if (newValue) {
+					sqlCmdVariables[sqlCmd.key] = newValue;
+				}
+			} else if (sqlCmd.isResetAllVars) {
+				sqlCmdVariables = Object.assign({}, project.sqlCmdVariables);
+			} else if (sqlCmd.isDone) {
+				break;
+			}
+
+		}
+	}
+
+	// 5. Select action to take
+	const action = await vscode.window.showQuickPick(
+		[constants.generateScriptButtonText, constants.publishDialogOkButtonText],
```

Comment:
update name of publishDialogOkButtonText to something more clear like publishButtonText

Reply:
FYI - in the future for loc strings I find it best to not give them names that imply their use and instead just give them names that refer to what the string actually is (so like this one I'm changing to just publish) which can then be re-used across anything that needs that string without it looking weird.